NeoMathEngine/include/NeoMathEngine/BlobDesc.h

```diff
@@ -120,29 +91,52 @@ class NEOMATHENGINE_API CBlobDesc {
 	bool HasEqualDimensions( const CBlobDesc& other ) const;
 	bool HasEqualDimensions( const CBlobDesc& other, std::initializer_list<int> dimensions ) const;
 	// Gets the dimensions of the blob
-	void GetDimSizes( int s[MaxDimensions] ) const
-	{
-		for( int i = 0; i < MaxDimensions; i++ ) { s[i] = dimensions[i]; }
-	}
+	void GetDimSizes( int s[MaxDimensions] ) const;
 
 	TBlobType GetDataType() const { return type; }
 	void SetDataType( TBlobType dataType ) { type = dataType; }
 
 private:
-	int dimensions[MaxDimensions];
-	TBlobType type;
+	int dimensions[MaxDimensions]{};
+	TBlobType type = CT_Invalid;
+};
+
+//---------------------------------------------------------------------------------------------------------------------
+
+inline CBlobDesc::CBlobDesc( TBlobType dataType ) :
+	type( dataType )
+{
+	for( int i = 0; i < MaxDimensions; i++ ) {
+		dimensions[i] = 1;
+	}
+}
+
+inline CBlobDesc::CBlobDesc( std::initializer_list<int> list ) :
+	type( CT_Float )
+{
+	int i = BD_Count - 1;
+	int j = static_cast<int>( list.size() ) - 1;
 
-	// The index of the dimension specified by name
-	static int getDimIndex( TBlobDim d )
-	{
-		return static_cast<int>( d );
+	while( i >= 0 && j >= 0 ) {
+		dimensions[i] = list.begin()[j];
+		i--;
+		j--;
 	}
 
-	static TBlobDim getDimName( int d )
-	{
-		return DimensionName[d];
+	while( i >= 0 ) {
+		dimensions[i] = 1;
+		i--;
 	}
-};
+}
```

Comment:
We can add `NeoAssert( j == 0, "Wrong dimensions list size" );`

Reply:
There is no include of `NeoAssert` in this header. And I think that nothing wrong to create an empty `CBlobDesc( {} );`, so someone can use it that way.